tests/python_package_test/test_sklearn.py

```diff
@@ -1149,3 +1139,49 @@ def test_check_is_fitted(self):
         rnk.fit(X, y, group=np.ones(X.shape[0]))
         for model in models:
             check_is_fitted(model)
+
+
+def _tested_estimators():
+    for Estimator in [lgb.sklearn.LGBMClassifier, lgb.sklearn.LGBMRegressor]:
+        yield Estimator()
+
+
+if sk_version < parse_version("0.23"):
+    def _generate_checks_per_estimator(check_generator, estimators):
+        for estimator in estimators:
+            name = estimator.__class__.__name__
+            for check in check_generator(name, estimator):
+                yield estimator, check
+
+    @pytest.mark.skipif(
+        sk_version < parse_version("0.21"), reason="scikit-learn version is less than 0.21"
+    )
+    @pytest.mark.parametrize(
+        "estimator, check",
+        _generate_checks_per_estimator(_yield_all_checks, _tested_estimators()),
+    )
+    def test_sklearn_integration(estimator, check):
+        xfail_checks = estimator._get_tags()["_xfail_checks"]
+        check_name = check.__name__ if hasattr(check, "__name__") else check.func.__name__
+        if xfail_checks and check_name in xfail_checks:
```

Comment:
With `get()` it looks a bit safer.
```suggestion
        xfail_checks = estimator._get_tags().get("_xfail_checks", {})
        check_name = check.__name__ if hasattr(check, "__name__") else check.func.__name__
        if check_name in xfail_checks:
```

Reply:
`"_xfail_checks"` is always defined because it is a default tag. At least, if there is a breaking change in the future, one could detect it with the failure. WDYT?

Reply:
Ah, so `_xfail_checks` is defined (with empty dict?) in BaseEstimator, if I got it correctly? Then it makes sense!

Then another question. Why do we need `if xfail_checks and ...` guard?